File: sdk/luma/sql.py
from typing import Any, List, Optional

from ._http import Http


class SqlClient:
    """SQLite bridge. Use query for SELECT, exec for DDL/DML."""

    def __init__(self, http: Http):
        self._http = http

    def query(self, sql: str, params: Optional[List[Any]] = None) -> dict:
        return self._http.post("/v1/sql/query", {"sql": sql, "params": params or []})

    async def aquery(self, sql: str, params: Optional[List[Any]] = None) -> dict:
        return await self._http.apost("/v1/sql/query", {"sql": sql, "params": params or []})

    def exec(self, sql: str, params: Optional[List[Any]] = None) -> dict:
        return self._http.post("/v1/sql/exec", {"sql": sql, "params": params or []})

    async def aexec(self, sql: str, params: Optional[List[Any]] = None) -> dict:
        return await self._http.apost("/v1/sql/exec", {"sql": sql, "params": params or []})

File: sdk/typescript/src/sql.ts
import { HttpClient } from "./http.js";
import type { JsonValue, SqlExecResult, SqlQueryResult } from "./types.js";

export class SqlClient {
  constructor(private readonly http: HttpClient) {}

  query(sql: string, params: JsonValue[] = []): Promise<SqlQueryResult> {
    return this.http.post("/v1/sql/query", { sql, params });
  }

  exec(sql: string, params: JsonValue[] = []): Promise<SqlExecResult> {
    return this.http.post("/v1/sql/exec", { sql, params });
  }
}
